great_expectations/render/renderer_configuration.py

```diff
@@ -1,43 +1,164 @@
+        exclude: Optional[Union[AbstractSetIntStr, MappingIntStrAny]] = None,
+        by_alias: bool = True,
+        skip_defaults: Optional[bool] = None,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+    ) -> DictStrAny:
+        """
+        Override BaseModel dict to make the default by_alias True instead of False.
+        We need by_alias to be set to True in RendererParams, because we have an existing
+        attribute named schema, and schema is already a Pydantic BaseModel attribute.
+        In practice this means the renderer implementer doesn't need to use
+        .dict(by_alias=True) everywhere.
+        """
+        return super().dict(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            skip_defaults=skip_defaults,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+        )
 
-@dataclass(frozen=True)
-class RendererConfiguration:
+
+class RendererConfiguration(BaseModel):
     """Configuration object built for each renderer."""
 
-    configuration: Union[ExpectationConfiguration, None]
-    result: Union[ExpectationValidationResult, None]
-    runtime_configuration: dict = field(default_factory=dict)
-    kwargs: dict = field(init=False)
-    include_column_name: bool = field(init=False)
-    styling: Union[dict, None] = field(init=False)
-
-    def __post_init__(self) -> None:
-        kwargs: dict
-        if self.configuration:
-            kwargs = self.configuration.kwargs
-        elif self.result and self.result.expectation_config:
-            kwargs = self.result.expectation_config.kwargs
-        else:
-            kwargs = {}
-
-        object.__setattr__(self, "kwargs", kwargs)
-
-        include_column_name: bool = True
-        styling: Union[dict, None] = None
-        if self.runtime_configuration:
-            include_column_name = (
+    configuration: Union[ExpectationConfiguration, None] = Field(
+        ..., allow_mutation=False
+    )
+    result: Union[ExpectationValidationResult, None] = Field(..., allow_mutation=False)
+    runtime_configuration: Union[dict, None] = Field({}, allow_mutation=False)
+    expectation_type: str = Field("", allow_mutation=False)
+    kwargs: dict = Field({}, allow_mutation=False)
+    include_column_name: bool = Field(True, allow_mutation=False)
+    styling: Union[dict, None] = Field(None, allow_mutation=False)
+    params: BaseModel = Field(default_factory=RendererParams, allow_mutation=True)
+
+    class Config:
+        validate_assignment = True
+        arbitrary_types_allowed = True
+
+    def __init__(self, **kwargs):
+        kwargs = RendererConfiguration._set_expectation_type_and_kwargs(kwargs=kwargs)
+        kwargs = RendererConfiguration._set_include_column_name_and_styling(
+            kwargs=kwargs
+        )
```

Comment:
@NathanFarmer have you looked at doing this with a custom validator?
https://docs.pydantic.dev/usage/validators/